msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IMDSRetryPolicy.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+//IMDS uses a different try policy than other MI flows, see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/docs/imds_retry_based_on_errors.md
+class IMDSRetryPolicy extends ManagedIdentityRetryPolicy {
+    private static final int LINEAR_RETRY_NUM = 7;
+    private static int LINEAR_RETRY_DELAY_MS = 10000; // 10 seconds
+    private static final int EXPONENTIAL_RETRY_NUM = 3;
+    private static int EXPONENTIAL_RETRY_DELAY_MS = 1000; // 1 second
+
+    private int currentRetryCount;
+    private int lastStatusCode;
+    
+    @Override
+    public boolean isRetryable(IHttpResponse httpResponse) {
+        currentRetryCount++;
+        lastStatusCode = httpResponse.statusCode();
+
+        return HttpStatus.isServerError(lastStatusCode) ||
+                lastStatusCode == HttpStatus.NOT_FOUND.getCode() ||
+                lastStatusCode == HttpStatus.REQUEST_TIMEOUT.getCode() ||
+                lastStatusCode == HttpStatus.GONE.getCode() ||
+                lastStatusCode == HttpStatus.TOO_MANY_REQUESTS.getCode();
```

Comment:
Not including `isServerError`, you should consider adding the other http status codes to an array, like in msal-node:

```
const HTTP_STATUS_400_CODES_FOR_EXPONENTIAL_STRATEGY: Array<number> = [
    HttpStatus.NOT_FOUND,
    HttpStatus.REQUEST_TIMEOUT,
    HttpStatus.GONE,
    HttpStatus.TOO_MANY_REQUESTS,
];
```

This is just a suggestion, you can ignore it.

Reply:
It's not as clean as node due to the (rather old) Java version that we support, however in the latest commit I did something similar: put all the appropriate codes in a static set, then just check if the set contains the response's status code.